### PR TITLE
feat: update email address

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

update email address flow 
https://wearezeta.atlassian.net/browse/AR-1073

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1679/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1679/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1679/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6277145)(6277145)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6277145)(6277145)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/verifyEmail/VerifyEmailScreen.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Preview
+@Composable
+fun PreviewChangeDisplayName() {
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/verifyEmail/VerifyEmailScreen.kt]:
```suggestion
fun PreviewVerifyEmailScreen() {
```

File: app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/updateEmail/ChangeEmailViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                    NavigationItem.VerifyEmailAddress.getRouteWithArgs(listOf(state.email.text)),
+                    BackStackMode.REMOVE_CURRENT
+                )
+            )
+        }
+    }
+
+    fun onEmailChange(newEmail: TextFieldValue) {
+        val cleanEmail = newEmail.text.trim()
+        val isValidEmail = cleanEmail.contains('@')
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/updateEmail/ChangeEmailViewModel.kt]: Question, is it possible to use something like this to validate ?https://developer.android.com/reference/android/util/Patterns#EMAIL_ADDRESS

Just asking without context, maybe the `@` Symbol is the requirement needed at Wire.

File: app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/updateEmail/ChangeEmailScreen.kt [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                stringResource(id = R.string.settings_myaccount_email_invalid_imail_error)
+            )
+        }
+    } else {
+        WireTextFieldState.Default
+    }
+
+@Preview
+@Composable
+fun PreviewChangeDisplayName() {
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/settings/account/email/updateEmail/ChangeEmailScreen.kt]:
```suggestion
fun PreviewChangeEmailScreen() {
```

File: app/src/main/kotlin/com/wire/android/ui/home/settings/account/MyAccountViewModel.kt [hunk 4]
```diff
@@ -67,41 +70,44 @@ class MyAccountViewModel @Inject constructor(
     var myAccountState by mutableStateOf(MyAccountState())
         private set
 
+    @VisibleForTesting
+    var hasSAMLCred by Delegates.notNull<Boolean>()
+
+    @VisibleForTesting
+    var managedByWire by Delegates.notNull<Boolean>()
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/settings/account/MyAccountViewModel.kt:77]: Life is not fair 😓  good solution to the problem of verifying these 2 conditions.

**Review by yamilmedina — COMMENTED** [review]
Great work, and refactors for some problems we had 🙌 
Just added some suggestions and questions.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1679/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6284466)(6284466)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6284466)(6284466)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1679/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6285003)(6285003)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6285003)(6285003)-staging-compat.apk)

**Review by yamilmedina — APPROVED** [review]
🔥

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-1679/11/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6286741)(6286741)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6286741)(6286741)-staging-compat.apk)

**Pull request merged**